PWGJE/Tasks/jetfragmentation.cxx

```diff
@@ -310,11 +512,86 @@ struct JetFragmentation {
+
+      registry.fill(HIST("data/V0/V0PtEtaPhi"), v0.pt(), v0.eta(), v0.phi());
+      registry.fill(HIST("data/V0/V0PtCtau"), v0.pt(), ctauK0s, ctauLambda, ctauAntiLambda);
+      registry.fill(HIST("data/V0/V0PtMass"), v0.pt(), v0.mK0Short(), v0.mLambda(), v0.mAntiLambda());
+      registry.fill(HIST("data/V0/V0PtRadiusCosPA"), v0.pt(), v0.v0radius(), v0.v0cosPA());
+      registry.fill(HIST("data/V0/V0PtDCAposneg"), v0.pt(), v0.dcapostopv(), v0.dcanegtopv());
+      registry.fill(HIST("data/V0/V0PtDCAd"), v0.pt(), v0.dcaV0daughters());
+
+      // auto negTrack = v0.template negTrack_as<TrackType>();
+      // auto posTrack = v0.template posTrack_as<TrackType>();
+      if (TMath::Abs(v0.dcanegtopv()) < dcanegtopv || TMath::Abs(v0.dcapostopv()) < dcapostopv || v0.dcaV0daughters() < dcav0dau) {
+        continue;
+      } // TODO: move to filter
+
+      if (v0.v0radius() > v0radius && v0.v0cosPA() > v0cospa) {
+        // TODO: Add TPC info to tracks
+        if (ctauLambda < lifetimecutLambda) {
+          // if (TMath::Abs(posTrack.tpcNSigmaPr()) < TpcPidNsigmaCut) {
+          registry.fill(HIST("data/V0/LambdaPtEtaPhi"), v0.pt(), v0.eta(), v0.phi());
+          registry.fill(HIST("data/V0/LambdaPtCtauMass"), v0.pt(), ctauLambda, v0.mLambda());
+          registry.fill(HIST("data/V0/LambdaPtRadiusCosPA"), v0.pt(), v0.v0radius(), v0.v0cosPA());
+          registry.fill(HIST("data/V0/LambdaPtDCAposneg"), v0.pt(), v0.dcapostopv(), v0.dcanegtopv());
+          registry.fill(HIST("data/V0/LambdaPtDCAd"), v0.pt(), v0.dcaV0daughters());
+          // }
+          // if (TMath::Abs(negTrack.tpcNSigmaPr()) < TpcPidNsigmaCut) {
+          registry.fill(HIST("data/V0/antiLambdaPtEtaPhi"), v0.pt(), v0.eta(), v0.phi());
+          registry.fill(HIST("data/V0/antiLambdaPtCtauMass"), v0.pt(), ctauAntiLambda, v0.mAntiLambda());
+          registry.fill(HIST("data/V0/antiLambdaPtRadiusCosPA"), v0.pt(), v0.v0radius(), v0.v0cosPA());
+          registry.fill(HIST("data/V0/antiLambdaPtDCAposneg"), v0.pt(), v0.dcapostopv(), v0.dcanegtopv());
+          registry.fill(HIST("data/V0/antiLambdaPtDCAd"), v0.pt(), v0.dcaV0daughters());
+          // }
+        } // Lambda lifetime cut
+        if (ctauK0s < lifetimecutK0S) {
+          registry.fill(HIST("data/V0/K0SPtEtaPhi"), v0.pt(), v0.eta(), v0.phi());
+          registry.fill(HIST("data/V0/K0SPtCtauMass"), v0.pt(), ctauK0s, v0.mK0Short());
+          registry.fill(HIST("data/V0/K0SPtRadiusCosPA"), v0.pt(), v0.v0radius(), v0.v0cosPA());
+          registry.fill(HIST("data/V0/K0SPtDCAposneg"), v0.pt(), v0.dcapostopv(), v0.dcanegtopv());
+          registry.fill(HIST("data/V0/K0SPtDCAd"), v0.pt(), v0.dcaV0daughters());
+        } // K0S lifetime cut
+      }   // v0 radius, cosPA
+    }     // for v0
+  }
+
+  template <typename CollisionType, typename JetType, typename V0Type, typename TrackType>
+  void fillDataV0FragHistograms(CollisionType collision, JetType jet, V0Type v0, TrackType tracks)
```

Comment:
why do you need tracks here?

Reply:
In the future, I will be using Tracks here for V0 PID. The tracks are not currently necessary here